pandas/core/reshape/util.py

```diff
@@ -66,17 +71,13 @@ def cartesian_product(X) -> list[np.ndarray]:
     ]
 
 
-def tile_compat(arr: NumpyIndexT, num: int) -> NumpyIndexT:
+def tile_compat(arr: AnyArrayLikeT, num: int, axis: AxisInt = 0) -> AnyArrayLikeT:
```

Comment:
do we ever get Series here?  i.e. can we tighten this?

Reply:
could do, but it does work with `Series` - this is just a generic internal `util` function, might as well keep it not too tight IMO